src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/VariantFmsDTOValidator.java

```diff
@@ -0,0 +1,325 @@
+import org.alliancegenome.curation_api.services.VocabularyTermService;
+import org.alliancegenome.curation_api.services.VocabularyTermSetService;
+import org.alliancegenome.curation_api.services.associations.alleleAssociations.AlleleVariantAssociationService;
+import org.alliancegenome.curation_api.services.associations.variantAssociations.CuratedVariantGenomicLocationAssociationService;
+import org.alliancegenome.curation_api.services.helpers.notes.NoteIdentityHelper;
+import org.alliancegenome.curation_api.services.helpers.variants.HgvsIdentifierHelper;
+import org.alliancegenome.curation_api.services.ontology.NcbiTaxonTermService;
+import org.alliancegenome.curation_api.services.ontology.SoTermService;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+
+import jakarta.enterprise.context.RequestScoped;
+import jakarta.inject.Inject;
+import jakarta.transaction.Transactional;
+
+@RequestScoped
+public class VariantFmsDTOValidator {
+
+	@Inject VariantDAO variantDAO;
+	@Inject NoteDAO noteDAO;
+	@Inject AlleleService alleleService;
+	@Inject AssemblyComponentDAO assemblyComponentDAO;
+	@Inject CuratedVariantGenomicLocationAssociationDAO curatedVariantGenomicLocationAssociationDAO;
+	@Inject CuratedVariantGenomicLocationAssociationService curatedVariantGenomicLocationAssociationService;
+	@Inject AlleleVariantAssociationDAO alleleVariantAssociationDAO;
+	@Inject AlleleVariantAssociationService alleleVariantAssociationService;
+	@Inject SoTermService soTermService;
+	@Inject DataProviderService dataProviderService;
+	@Inject NcbiTaxonTermService ncbiTaxonTermService;
+	@Inject VocabularyTermService vocabularyTermService;
+	@Inject VocabularyTermSetService vocabularyTermSetService;
+	@Inject CrossReferenceFmsDTOValidator crossReferenceFmsDtoValidator;
+	@Inject CrossReferenceService crossReferenceService;
+	@Inject VariantNoteFmsDTOValidator variantNoteFmsDtoValidator;
+	
+	@Transactional
+	public Variant validateVariant(VariantFmsDTO dto, List<Long> idsAdded, BackendBulkDataProvider dataProvider) throws ValidationException {
+
+		ObjectResponse<Variant> variantResponse = new ObjectResponse<Variant>();
+		Variant variant = new Variant();
+		
+		if (dto.getStart() == null) {
+			variantResponse.addErrorMessage("start", ValidationConstants.REQUIRED_MESSAGE);
+		}
+		
+		if (dto.getEnd() == null) {
+			variantResponse.addErrorMessage("end", ValidationConstants.REQUIRED_MESSAGE);
+		}
+		
+		if (StringUtils.isBlank(dto.getSequenceOfReferenceAccessionNumber())) {
+			variantResponse.addErrorMessage("sequenceOfReferenceAccessionNumber", ValidationConstants.REQUIRED_MESSAGE);
+		}
+		
+		SOTerm variantType = null;
+		if (StringUtils.isBlank(dto.getType())) {
+			variantResponse.addErrorMessage("type", ValidationConstants.REQUIRED_MESSAGE);
+		} else if (Objects.equals(dto.getType(), "SO:1000002") || Objects.equals(dto.getType(), "SO:1000008") ||
```

Comment:
🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.OperatorWrapCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'||' should be on a new line.
<!-- __reviewdog__:ChA4ZmFmNTY1MzE0MjVhODMyEgpjaGVja3N0eWxl -->